### PR TITLE
Feature/rating screen and data

### LINKED ISSUE #129: Add Student feedback and ratings to the Tutor, so that Student can see previous Students experiences. "TutorInfoScreenForStudents"

### PR DESCRIPTION
**Pull request by alexs-rufer**
## 🎯 PR Description

I added a `rating` field for tutor profiles to enable the storage of ratings and comments for each tutor. This feature allows students to review tutors, providing an average rating, total number of ratings, and a list of comments. The rating information is now retrieved from and stored in Firestore. I also spent the afternoon writing the corresponding Screen (Display tutor ratings, when clicking on a tutor card) which I almost finished, then realising @LeaBlancher had already finished this screen. I don't know if I misunderstood my task or not but I'll try to communicate more next time. Thus I am only pushing the Rating implementation and not the screen.

## 📂 Files changed

- **Comment.kt**
  - Created the `Comment` data class to represent individual reviews. Each comment includes:
    - `grade` (1 to 5 stars), `raterUid` (the UID of the student who left the review), `date` (the timestamp when the review was left), `comment` (the content of the review).

- **Rating.kt**
  - Created the `Rating` data class to represent tutor ratings:
    - Fields include `averageRating`, `totalRatings`, and a mutable list of `Comment` objects.
    - Added a `toMap` method to serialize the `Rating` object into a map for Firestore.
    - Added a `mapToRating` method to deserialize Firestore data into a `Rating` object.
    - Added an `addRating` method to dynamically add new comments and update the average rating.
   
- **Profile.kt**
  - Added a new `rating` field to the `Profile` data class with a default `Rating()` object.

- **ProfilesRepositoryFirestore.kt**
  - Updated `documentToProfile`:
    - Now retrieves the `rating` field from Firestore documents as a map and converts it into a `Rating` object using the `Rating.mapToRating` helper function.
  - Updated `profileToMap`:
    - Added the `rating` field to the map representation of `Profile`, converting it using the `Rating.toMap` method for Firestore compatibility.

## 🧪 How to test
There is no good way to test the implementation since we did not implemented yet the feature to add rating and the screen of @LeaBlancher to display them. I did some very (very) basic testing. I did a workaround by adding a review and 2 comments in the overview screen verifying that it correctly updates the firebase. Then I tested the other way around by logging the value of the profile to make sure the ratings are deserialised correctly, which also worked.

Close #129

### CODE REVIEW

File: app/src/main/java/com/github/se/project/model/profile/Comment.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.github.se.project.model.profile
+
+import com.google.firebase.Timestamp
+
+data class Comment(
+    val grade: Int, // rating given (1 to 5 stars)
+    val raterUid: String, // uid of the person who rated
+    val date: Timestamp, // date of the rating
+    val comment: String // actual comment
+)
```
Comment (LeaBlancher) [inline app/src/main/java/com/github/se/project/model/profile/Comment.kt:10]: This data class is very well thought to group all the usefull fields for the comment! Good job!

File: app/src/main/java/com/github/se/project/model/profile/Rating.kt [hunk 2]
```diff
@@ -0,0 +1,65 @@
+package com.github.se.project.model.profile
+
+import com.google.firebase.Timestamp
+
+data class Rating(
+    var averageRating: Double = 0.0, // average rating (1 to 5 stars)
+    var totalRatings: Int = 0, // total number of ratings
+    val comments: MutableList<Comment> = mutableListOf() // list of comments
```
Comment (LeaBlancher) [inline app/src/main/java/com/github/se/project/model/profile/Rating.kt:8]: Also a very useful class! Thanks

File: app/src/main/java/com/github/se/project/model/profile/Rating.kt [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      val comments =
+          (ratingMap["comments"] as? List<Map<String, Any?>>)?.map { commentMap ->
+            Comment(
+                grade = (commentMap["grade"] as? Long)?.toInt() ?: 0,
+                raterUid = commentMap["raterUid"] as? String ?: "",
+                date = commentMap["date"] as? Timestamp ?: Timestamp.now(),
+                comment = commentMap["comment"] as? String ?: "")
+          } ?: emptyList()
+
+      return Rating(averageRating, totalRatings, comments.toMutableList())
```
Comment (LeaBlancher) [inline app/src/main/java/com/github/se/project/model/profile/Rating.kt:62]: Good job, it seems very clear and works well for deserialization.

**Review by LeaBlancher — APPROVED** [review]
You PR is really good, the code is well documented as well as the PR is so it's very helpful for reviews (now and later on)! The add of rating and comments is very well done with the class you created that are very complete and well thought. The serialization and deserialization are also very well handled. Good job!
Sorry again for the misunderstood of the screen, we will make sure to better communicate in the future...

File: app/src/androidTest/java/com/github/se/project/ui/lesson/TutorMatchingTest.kt [hunk 4]
```diff
@@ -62,7 +62,7 @@ import androidx.compose.ui.test.*
                   academicLevel = AcademicLevel.MA1,
                   languages = listOf(Language.ENGLISH),
                   subjects = listOf(Subject.ANALYSIS),
-                  schedule = listOf(),
+                  schedule = List(7) { List(12) { 1 } },
```
Comment (LeaBlancher) [inline app/src/androidTest/java/com/github/se/project/ui/lesson/TutorMatchingTest.kt:65]: Thank you for catching these bugs in the test! I don't even know how it passed before.

**Comment by alexs-rufer** [comment]
I merged the main with the commented tests that make the CI crash, so that the CI is ok. this is a decision that was taken with the whole group.

**Review by AlexisAllmnd — APPROVED** [review]
Approve the merged version with the main

**Review by LeaBlancher — APPROVED** [review]
Their was only small changes to merge the main and correct the test. It is still okay to be merged. Good job!

**Pull request merged**